### PR TITLE
DOP-3159: README overhaul

### PR DESCRIPTION
**Pull request by seungpark**
[DOP-3159](https://jira.mongodb.org/browse/DOP-3159)

- update parser readme
- include details for prerequisites and steps to install
- include details for granular components
- include expected input of AST
- minor additions to add intention of packages

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -67,10 +77,16 @@ helper classes.
 
 #### `flutter.py`
 
+Tool to verify and mark validity for any Giza-style YAML input.
+
 #### `types.py`
 
+Definitions of class components that make up the parser. See n.py for more granular types of nodes that are expected within the reStructuredText.
```
Comment (i80and) [inline HACKING.md:84]: I actually think `n.py` is worth having its own section!

File: HACKING.md [hunk 2]
```diff
@@ -67,10 +77,16 @@ helper classes.
 
 #### `flutter.py`
 
+Tool to verify and mark validity for any Giza-style YAML input.
```
Comment (i80and) [inline HACKING.md:80]: More generally, flutter is a tool that loads arbitrary JSON/YAML/TOML data into a class and reports schema violations. It's also used for configuration files and such, not just Giza files!

File: HACKING.md [hunk 3]
```diff
@@ -38,7 +38,19 @@ for use with IDEs such as Visual Studio Code.
...
+the parsed directory is as follows:
+```
+reStructuredText directory
+┣snooty.toml
+┣source
+┃ ┗images
+┃   ┗image files here
+┃ ┗page-name
+┃   ┗sub-page-name.txt
+┃ ┗index.txt
```
Comment (i80and) [inline HACKING.md:51]: bonus points if you include
```
┃ ┗includes
┃   ┗fact-some-content.rst
```

**Review by i80and — COMMENTED** [review]
Good stuff!

File: HACKING.md (hunk 1 above)
Comment (seungpark) [reply HACKING.md]: done!

File: HACKING.md (hunk 2 above)
Comment (seungpark) [reply HACKING.md]: good catch! thanks for the clarification. updated

**Review by i80and — APPROVED** [review]
🙌🏻

**Pull request merged**